pkg/controllers/clusterinventory/clusterprofile/controller.go

```diff
@@ -0,0 +1,265 @@
+		return ctrl.Result{}, err
+	}
+	klog.V(2).InfoS("Cluster profile object is created or updated", "MemberCluster", mcRef, "ClusterProfile", klog.KObj(cp), "operation", createOrUpdateRes)
+	// sync the cluster profile condition from the member cluster condition
+	r.syncClusterProfileCondition(mc, cp)
+	if err = r.Status().Update(ctx, cp); err != nil {
+		klog.ErrorS(err, "Failed to update cluster profile status", "MemberCluster", mcRef, "ClusterProfile", klog.KObj(cp))
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// syncClusterProfileCondition syncs the ClusterProfile object's condition based on the MemberCluster object's condition.
+func (r *Reconciler) syncClusterProfileCondition(mc *clusterv1beta1.MemberCluster, cp *clusterinventory.ClusterProfile) {
+	// Update the cluster profile status.
+	//
+	// For simplicity reasons, for now only the health check condition is populated, using
+	// Fleet member agent's API server health check result.
+	var mcHealthCond *metav1.Condition
+	var memberAgentLastHeartbeat *metav1.Time
+
+	memberAgentStatus := mc.GetAgentStatus(clusterv1beta1.MemberAgent)
+	if memberAgentStatus != nil {
+		mcHealthCond = meta.FindStatusCondition(memberAgentStatus.Conditions, string(clusterv1beta1.AgentHealthy))
+		memberAgentLastHeartbeat = &memberAgentStatus.LastReceivedHeartbeat
+	}
+	switch {
+	case memberAgentStatus == nil:
+		// The member agent hasn't reported its status yet.
+		// Set the unknown health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
+			Reason:             clusterNoStatusReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent has not reported its status yet",
+		})
+	case mcHealthCond == nil:
+		// The member agent has reported its status, but the health condition is missing.
+		// Set the unknown health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
+			Reason:             clusterHealthUnknownReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent has reported its status, but the health condition is missing",
+		})
+	case memberAgentLastHeartbeat == nil || time.Since(memberAgentLastHeartbeat.Time) > r.ClusterUnhealthyThreshold:
+		// The member agent has lost its heartbeat connection to the Fleet hub cluster.
+		// Set the unknown health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
```

Comment:
should it be false?

Reply:
The condition is "ClusterConditionControlPlaneHealthy". So semanticallym, it seems that it should be unknown as we lost the heartbeat so we don't know.